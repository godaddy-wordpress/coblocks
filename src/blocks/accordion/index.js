/**
 * External dependencies
 */
import { AccordionIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies
 */
import edit from './edit';
import metadata from './block.json';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';
import { InnerBlocks } from '@wordpress/block-editor';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	attributes,
	/* translators: block description */
	description: __( 'Organize content within collapsable accordion items.', 'coblocks' ),
	edit,
	example: {
		attributes: {
			count: 2,
		},
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'tabs', 'coblocks' ),
		/* translators: block keyword (abbreviation for "frequently asked questions") */
		__( 'faq', 'coblocks' ),
	],
	save() {
		return (
			<div>
				<InnerBlocks.Content />
			</div>
		);
	},
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
	},
	/* translators: block name */
	title: __( 'Accordion', 'coblocks' ),
	transforms,
};

export { name, category, metadata, settings };
