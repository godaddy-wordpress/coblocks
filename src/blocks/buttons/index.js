/**
 * THIS BLOCK IS DEPRECRATED. IT WILL NOT SHOW UP IN THE INSERTER ANYMORE.
 * SEE /src/js/deprecations/deprecate-coblocks-buttons.js
 */

/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { button as icon } from '@wordpress/icons';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Buttons', 'coblocks' ),
	/* translators: block description */
	description: __( 'Prompt visitors to take action with multiple buttons, side by side.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'link', 'coblocks' ),
		/* translators: block keyword (abbreviation for "call to action") */
		__( 'cta', 'coblocks' ),
		/* translators: block keyword */
		__( 'call to action', 'coblocks' ),
	],
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings };
