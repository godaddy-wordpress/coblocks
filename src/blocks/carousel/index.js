/**
 * WordPress dependencies
 */
import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';

/**
 * Internal dependencies
 */
import BlockEdit from './block-edit';
import metadata from './block.json';

const settings = {
	edit: BlockEdit,
	save: () => (
		<div { ...useBlockProps.save() }>
			<InnerBlocks.Content />
		</div>
	),
};

const name = metadata.name;
export { name, metadata, settings };
