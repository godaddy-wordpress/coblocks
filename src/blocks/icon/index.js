/**
 * External dependencies
 */
import { IconIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import { hasFormattingCategory } from '../../utils/block-helpers';
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';

/**
 * Set default icon size equivalent to "Medium".
 */
export const DEFAULT_ICON_SIZE = 60;

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	attributes,
	category: hasFormattingCategory ? 'common' : 'design',
	deprecated,
	/* translators: block description */
	description: __( 'Add a stylized graphic symbol to communicate something more.', 'coblocks' ),
	edit,
	// We can't enable example on this block without breaking style previews as attributes are completely replaced by the example.
	//example: {}
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		'svg',
		/* translators: block keyword */
		__( 'icons', 'coblocks' ),
	],
	save: () => null,
	styles: [
		{
			isDefault: true,
			/* translators: block style */
			label: __( 'Outlined', 'coblocks' ),
			name: 'outlined',
		},
		{
			/* translators: block style */
			label: __( 'Filled', 'coblocks' ),
			name: 'filled',
		},
	],
	/* translators: block name */
	title: __( 'Icon', 'coblocks' ),
};

export { name, category, metadata, settings };
