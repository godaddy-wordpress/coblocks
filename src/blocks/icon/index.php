<?php
/**
 * Server-side rendering of the `coblocks/icon` block.
 *
 * @package CoBlocks
 */

/**
 * Renders the block on server.
 *
 * @param array $attrs The block attributes.
 *
 * @return string Returns the block content.
 */
function coblocks_render_icon_block( $attrs ) {
	if ( ! is_array( $attrs ) ) {
		$attrs = array();
	}

	$wrapper_classes = array(
		'wp-block-coblocks-icon',
		! empty( $attrs['contentAlign'] ) ? "has-text-align-{$attrs['contentAlign']}" : false,
		! empty( $attrs['className'] ) ? $attrs['className'] : false,
	);

	$icon_color_class = ! empty( $attrs['iconColor'] )
		? _wp_to_kebab_case( $attrs['iconColor'] )
		: false;

	$background_color_class = ! empty( $attrs['backgroundColor'] )
		? _wp_to_kebab_case( $attrs['backgroundColor'] )
		: false;

	$inner_classes = array(
		'wp-block-coblocks-icon__inner',
		$background_color_class ? "has-{$background_color_class}-background-color" : false,
		( ! empty( $attrs['backgroundColor'] ) || ! empty( $attrs['customBackgroundColor'] ) ) ? 'has-background' : false,
		( ! empty( $attrs['iconColor'] ) || ! empty( $attrs['customIconColor'] ) ) ? 'has-text-color' : false,
		$icon_color_class ? "has-{$icon_color_class}-color" : false,
	);

	$color_styles = array(
		! empty( $attrs['customBackgroundColor'] ) ? "background-color: {$attrs['customBackgroundColor']};" : false,
		! empty( $attrs['customIconColor'] ) ? "color: {$attrs['customIconColor']}; fill: {$attrs['customIconColor']};" : false,
	);

	$inner_styles = array(
		! empty( $attrs['borderRadius'] ) ? "border-radius: {$attrs['borderRadius']}px;" : false,
		! empty( $attrs['height'] ) ? "height: {$attrs['height']}px;" : false,
		! empty( $attrs['padding'] ) ? "padding: {$attrs['padding']}px;" : false,
		! empty( $attrs['width'] ) ? "width: {$attrs['width']}px;" : false,
	);

	// If there is no anchor put the color styles on the outer div.
	if ( empty( $attrs['href'] ) ) {
		$inner_styles = array_merge( $inner_styles, $color_styles );
	}

	$color_styles    = implode( ' ', array_filter( $color_styles ) );
	$wrapper_classes = implode( ' ', array_filter( $wrapper_classes ) );
	$inner_classes   = implode( ' ', array_filter( $inner_classes ) );
	$inner_styles    = implode( ' ', array_filter( $inner_styles ) );

	// Icon fetch.
	$icon_style       = '-outlined';
	$icon_name        = _wp_to_kebab_case( $attrs['icon'] );
	$icon_path        = COBLOCKS_PLUGIN_DIR . "src/blocks/icon/svgs/$icon_name";
	$custom_icon_path = get_stylesheet_directory() . "/coblocks/icons/$icon_name";
	$icon             = '';

	if ( isset( $attrs['className'] ) && false !== strpos( $attrs['className'], 'is-style-filled' ) ) {
		// Filled doesn't have an extension. Only outlined does. Legacy behavior.
		$icon_style = '';
		$icon_name  = $attrs['icon'];
	}

	$custom_icon_path = is_readable( "$custom_icon_path$icon_style.svg" ) ? "$custom_icon_path$icon_style.svg" : $custom_icon_path . '.svg';
	$icon_path        = is_readable( "$icon_path$icon_style.svg" ) ? "$icon_path$icon_style.svg" : $icon_path . '.svg';

	if ( is_readable( $custom_icon_path ) ) {
		// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
		$icon = file_get_contents( $custom_icon_path );
	} elseif ( is_readable( $icon_path ) ) {
		// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
		$icon = file_get_contents( $icon_path );
	}

	if ( ! empty( $attrs['href'] ) ) {
		$icon = sprintf(
			'<a href="%1$s" rel="%2$s" target="%3$s" style="%4$s">%5$s</a>',
			$attrs['href'],
			! empty( $attrs['rel'] ) ? esc_attr( $attrs['rel'] ) : '',
			! empty( $attrs['linkTarget'] ) ? esc_attr( $attrs['linkTarget'] ) : '_self',
			esc_attr( $color_styles ), // To make sure the color gets to the svg when there is an anchor.
			$icon
		);
	}

	return trim(
		sprintf(
			'
			<div class="%1$s">
				<div class="%2$s" style="%3$s">
					%4$s
				</div>
			</div>
			',
			esc_attr( $wrapper_classes ),
			esc_attr( $inner_classes ),
			esc_attr( $inner_styles ),
			$icon
		)
	);
}

/**
 * Registers the block on server.
 */
function coblocks_register_icon_block() {
	if ( ! function_exists( 'register_block_type' ) ) {
		return;
	}

	register_block_type(
		COBLOCKS_PLUGIN_DIR . 'src/blocks/icon',
		array(
			'render_callback' => 'coblocks_render_icon_block',
		)
	);
}
add_action( 'init', 'coblocks_register_icon_block' );
