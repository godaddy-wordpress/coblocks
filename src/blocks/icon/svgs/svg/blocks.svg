<svg fill="none" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="m10.3519669 11.7141378v8.2858622h8.2815735v-8.2858622zm-10.3519669 8.2858622h8.2815735v-8.2858622h-8.2815735zm0-18.64319006v8.28586225h8.2815735v-8.28586225zm14.1407867-1.35680994-5.8592132 5.85189021 5.8592132 5.86224759 5.8592133-5.86224759z" fill="currentColor" /></svg>