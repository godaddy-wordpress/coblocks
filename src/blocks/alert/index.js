/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const { replaceBlocks } = dispatch( 'core/block-editor' );

		replaceBlocks(
			[ props.clientId ],
			createBlock( 'core/paragraph', props.attributes ),
		);
		return null;
	},
	parent: [],
	save: () => null,
	title: metadata.title,
};

export { name, metadata, settings };
