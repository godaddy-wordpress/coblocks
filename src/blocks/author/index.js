/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch, select } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const parentBlock = select( 'core/editor' ).getBlocksByClientId( props.clientId )[ 0 ];
		if ( ! parentBlock ) {
			return null;
		}

		// Used to set text color props
		const hasTextColor = ( props.attributes.hasOwnProperty( 'textColor' ) );
		const hasCustomTextColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'text' ) );

		// Used to set background color props
		const hasBackgroundColor = ( props.attributes.hasOwnProperty( 'backgroundColor' ) );
		const hasCustomBackgroundColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'background' ) );

		// Used to set font props
		const hasFontSize = ( props.attributes.hasOwnProperty( 'fontSize' ) );
		const hasCustomFont = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'typography' ) );

		const { replaceBlocks } = dispatch( 'core/block-editor' );
		const isRTL = select( 'core/editor' ).getEditorSettings().isRTL;

		const authorNameBlock = () => {
			let authorNameBlockProps = {
				content: props.attributes.name,
				level: 4,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			// Headings do not inherit color from parent - we descend the style here for initial match.
			if ( hasCustomTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						color: { ...authorNameBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			return createBlock( 'core/paragraph', authorNameBlockProps );
		};

		const authorBioBlock = () => {
			let authorBioBlockProps = {
				content: props.attributes.biography,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					style: {
						...authorBioBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			return createBlock( 'core/paragraph', authorBioBlockProps );
		};

		const buttonBlock = parentBlock.innerBlocks;
		// Set buttons to align.
		buttonBlock.forEach( function( button, index ) {
			buttonBlock[ index ].attributes.align = isRTL ? 'right' : 'left';
		} );
		// Button blocks are no-longer stand-alone and should be within `core/buttons` block.
		const coreButtons = createBlock( 'core/buttons', { }, [ ...buttonBlock ] );

		const imageBlock = createBlock( 'core/image', { className: 'is-style-rounded', id: props.attributes.imgId, url: props.attributes.imgUrl } );

		const leftColumn = createBlock( 'core/column', { width: '25%' }, [ imageBlock ] );
		const rightColumn = createBlock( 'core/column', { width: '75%' }, [ authorNameBlock(), authorBioBlock(), coreButtons ] );

		const columnsBlock = () => {
			let columnsBlockProps = {
				...( props.attributes.hasOwnProperty( 'className' ) && { className: props.attributes.className } ),
				...( props.attributes.hasOwnProperty( 'animation' ) && { animation: props.attributes.animation } ),
				...( props.attributes.hasOwnProperty( 'anchor' ) && { anchor: props.attributes.anchor } ),
				style: {
					spacing: {
						padding: {
							bottom: '2.5rem',
							left: '2.5rem',
							right: '2.5rem',
							top: '2.5rem',
						},
					},
				},
			};

			// Migrate text colors into columns block props.
			if ( hasCustomTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			// Migrate background colors into columns block props.
			if ( hasCustomBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: props.attributes.style.color.background,
						},
					},
				};
			}
			if ( hasBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					backgroundColor: props.attributes.backgroundColor,
				};
			}

			// Does not have background color set. Use default block color from CoBlocks.
			const hasNoBackgroundColorSet = ! props.attributes.hasOwnProperty( 'backgroundColor' ) &&
				! props.attributes?.style?.color?.hasOwnProperty( 'background' );
			if ( hasNoBackgroundColorSet ) {
				// $dark-opacity-light-200 computed #8C8C971A - no variable access in root.
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: '#8C8C971A',
						},
					},
				};
			}

			return createBlock( 'core/columns', columnsBlockProps, [ leftColumn, rightColumn ] );
		};

		replaceBlocks(
			[ props.clientId ],
			columnsBlock()
		);

		return null;
	},
	parent: [],
	save: () => null,
	/* translators: block name */
	title: metadata.title,
};

export { name, metadata, settings };
