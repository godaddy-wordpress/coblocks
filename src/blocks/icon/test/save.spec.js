/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { registerBlockType, createBlock, serialize } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;

describe( 'coblocks/icon', () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );
	} );

	it( 'should render', () => {
		block.attributes.icon = 'coblocks';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '"icon":"coblocks"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with height and width', () => {
		block.attributes.height = '200';
		block.attributes.width = '200';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '"height":"200"' );
		expect( serializedBlock ).toContain( '"width":"200"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with borderRadius and padding', () => {
		block.attributes.borderRadius = '200';
		block.attributes.padding = '30';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '"borderRadius":"200"' );
		expect( serializedBlock ).toContain( '"padding":"30"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with href, rel, and linkTarget', () => {
		block.attributes.href = 'https://wordpress.org';
		block.attributes.rel = 'rel';
		block.attributes.linkTarget = '_blank';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '"href":"https://wordpress.org"' );
		expect( serializedBlock ).toContain( '"rel":"rel"' );
		expect( serializedBlock ).toContain( '"linkTarget":"_blank"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with iconColor', () => {
		block.attributes.iconColor = 'primary';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '{"iconColor":"primary"}' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with customIconColor', () => {
		block.attributes.customIconColor = '#da5d5d';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '{"customIconColor":"#da5d5d"}' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with backgroundColor', () => {
		block.attributes.backgroundColor = 'primary';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '{"backgroundColor":"primary"}' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with customBackgroundColor', () => {
		block.attributes.customBackgroundColor = '#da5d5d';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '{"customBackgroundColor":"#da5d5d"}' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with className', () => {
		block.attributes.className = 'my-custom-class is-style-filled';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '{"className":"my-custom-class is-style-filled"}' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with an icon', () => {
		block.attributes.icon = 'coblocks';
		block.attributes.href = 'https://www.godaddy.com';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '"icon":"coblocks"' );
		expect( serializedBlock ).toContain( 'https://www.godaddy.com' );
		expect( serializedBlock ).toMatchSnapshot();
	} );
} );
