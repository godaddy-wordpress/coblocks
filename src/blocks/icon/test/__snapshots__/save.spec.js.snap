// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/icon should render 1`] = `"<!-- wp:coblocks/icon {\\"icon\\":\\"coblocks\\"} /-->"`;

exports[`coblocks/icon should render with an icon 1`] = `"<!-- wp:coblocks/icon {\\"icon\\":\\"coblocks\\",\\"href\\":\\"https://www.godaddy.com\\"} /-->"`;

exports[`coblocks/icon should render with backgroundColor 1`] = `"<!-- wp:coblocks/icon {\\"backgroundColor\\":\\"primary\\"} /-->"`;

exports[`coblocks/icon should render with borderRadius and padding 1`] = `"<!-- wp:coblocks/icon {\\"borderRadius\\":\\"200\\",\\"padding\\":\\"30\\"} /-->"`;

exports[`coblocks/icon should render with className 1`] = `"<!-- wp:coblocks/icon {\\"className\\":\\"my-custom-class is-style-filled\\"} /-->"`;

exports[`coblocks/icon should render with customBackgroundColor 1`] = `"<!-- wp:coblocks/icon {\\"customBackgroundColor\\":\\"#da5d5d\\"} /-->"`;

exports[`coblocks/icon should render with customIconColor 1`] = `"<!-- wp:coblocks/icon {\\"customIconColor\\":\\"#da5d5d\\"} /-->"`;

exports[`coblocks/icon should render with height and width 1`] = `"<!-- wp:coblocks/icon {\\"height\\":\\"200\\",\\"width\\":\\"200\\"} /-->"`;

exports[`coblocks/icon should render with href, rel, and linkTarget 1`] = `"<!-- wp:coblocks/icon {\\"href\\":\\"https://wordpress.org\\",\\"rel\\":\\"rel\\",\\"linkTarget\\":\\"_blank\\"} /-->"`;

exports[`coblocks/icon should render with iconColor 1`] = `"<!-- wp:coblocks/icon {\\"iconColor\\":\\"primary\\"} /-->"`;
